src/System.Security.Cryptography.Encoding/tests/Asn1/Writer/WriteCharacterString.cs

```diff
@@ -494,5 +494,57 @@ protected void VerifyWrite_Span_NonEncodable(string input)
                 Assert.Throws<EncoderFallbackException>(() => WriteSpan(writer, input.AsSpan()));
             }
         }
+
+        protected void WriteAfterDispose_Span(bool empty)
+        {
+            using (AsnWriter writer = new AsnWriter(AsnEncodingRules.DER))
+            {
+                if (!empty)
+                {
+                    writer.WriteNull();
+                }
+
+                writer.Dispose();
+
+#if netcoreapp
+                string input = "1";
+
+                Assert.Throws<ObjectDisposedException>(
+                    () => WriteSpan(writer, input));
```

Comment:
If this was `WriteSpan(writer, input.AsSpan())` would that remove the need for the conditional compile?